studio/src/mas-create-dialog.js

```diff
@@ -0,0 +1,205 @@
+import { editFragment } from './store.js';
+import { Fragment } from './aem/fragment.js';
+
+export class MasCreateDialog extends LitElement {
+    static properties = {
+        type: { type: String, reflect: true },
+        title: { state: true },
+        name: { state: true },
+        nameModified: { state: true },
+    };
+
+    static styles = css`
+        .form-field {
+            margin-bottom: var(--spectrum-global-dimension-size-400);
+        }
+
+        sp-field-label {
+            display: block;
+            margin-bottom: var(--spectrum-global-dimension-size-100);
+        }
+
+        sp-picker,
+        sp-textfield {
+            width: 100%;
+        }
+    `;
+
+    constructor() {
+        super();
+        this.type = 'merch-card';
+        this.title = '';
+        this.name = '';
+        this.nameModified = false;
+
+        // Bind methods to ensure correct 'this' context
+        this.handleTitleChange = this.handleTitleChange.bind(this);
+        this.handleNameChange = this.handleNameChange.bind(this);
+        this.handleNameFocus = this.handleNameFocus.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.close = this.close.bind(this);
+        this.handleKeyDown = this.handleKeyDown.bind(this);
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+        document.addEventListener(EVENT_KEYDOWN, this.handleKeyDown);
+    }
+
+    disconnectedCallback() {
+        super.disconnectedCallback();
+        document.removeEventListener(EVENT_KEYDOWN, this.handleKeyDown);
+    }
+
+    handleKeyDown(event) {
+        if (event.key === 'Escape' && this.dialog?.open) {
+            this.close();
+        }
+    }
+
+    get dialog() {
+        return this.shadowRoot.querySelector('sp-dialog-wrapper');
+    }
+
+    updated() {
+        this.open();
+    }
+
+    async open() {
+        await this.updateComplete;
+        this.dialog.open = true;
+    }
+
+    /**
+     * Normalizes a string to be used as a fragment name
+     * Converts to lowercase, replaces spaces with hyphens, and removes special characters
+     * @param {string} str - The string to normalize
+     * @returns {string} - The normalized string
+     */
+    normalizeFragmentName(str) {
+        if (!str) return '';
+        return str
+            .toLowerCase()
+            .replace(/\s+/g, '-')
+            .replace(/[^a-z0-9-]/g, '')
+            .replace(/-+/g, '-')
+            .replace(/^-|-$/g, '');
+    }
+
+    handleTitleChange(value) {
+        this.title = value;
+        // Only update name if it hasn't been modified by the user
+        if (!this.nameModified) {
+            this.name = this.normalizeFragmentName(value);
+        }
+    }
+
+    handleNameChange(value) {
+        this.name = value;
+        this.nameModified = !!value;
+    }
+
+    handleNameFocus() {
+        this.nameModified = true;
+    }
+
+    async handleSubmit(event) {
```

Comment:
We should probably add error handling here with try/catch.

Reply:
a toast message is displayed by mas-repository already.